Packs/GenerateAsBuilt/Scripts/GenerateAsBuilt/GenerateAsBuilt.py

```diff
@@ -0,0 +1,890 @@
+                "id": f"{playbook_id}",
+                "type": "playbook"
+            }
+        ],
+        "dependencyLevel": "optional"
+    }
+    dependencies = post_api_request(DEMISTO_DEPENDENCIES_PATH, body).get("existing").get("playbook").get(playbook_id)
+    if not dependencies:
+        return_error(f"Failed to retrieve dependencies for {playbook_id}")
+
+    types = {}
+    for dependency in dependencies:
+        d_type = dependency.get("type")
+        if d_type not in types:
+            types[d_type] = []
+
+        pack = dependency.get("packID", "Custom")
+        if not pack:
+            pack = "Custom"
+
+        types[d_type].append({
+            "type": d_type,
+            "name": dependency.get("name"),
+            "pack": pack
+        })
+
+    result_table_datas = {}
+    for k, v in types.items():
+        if v:
+            td = SortedTableData(v, f"{k}s", sort_key="name")
+            result_table_datas[k] = td
+
+    return result_table_datas
+
+
+def get_custom_automations():
+    r = post_api_request(DEMISTO_AUTOMATIONS_PATH, {"query": "system:F AND hidden:F"}).get("scripts")
+    rd = SortedTableData(r, "Custom Automations", "name")
+    return rd
+
+
+def get_system_config():
+    r = get_api_request(DEMISTO_CONFIG_PATH).get("defaultMap")
+    rd = TableData(r, "System Configuration")
+    return rd
+
+
+def main():
+    if demisto.args().get("playbook"):
+        # If we get a playbook, we generate a use case document, instead of teh platform as build
+        r = get_playbook_dependencies(demisto.args().get("playbook"))
+        doc = UseCaseDocument(
+            playbook_name=demisto.args().get("playbook"),
+            dependencies=r
+        )
+        fr = fileResult("usecase.html", doc.html())
+        return_results(fr)
+        return_results(CommandResults(
+            readable_output=doc.markdown(),
+        ))
+        return
+
+    # If no playbook is passed, we generate a platform as built.
+    open_incidents = get_open_incidents(MAX_DAYS, MAX_REQUEST_SIZE)
+    closed_incidents = get_closed_incidents(MAX_DAYS, MAX_REQUEST_SIZE)
+
+    system_config = get_system_config()
+    integrations = get_enabled_integrations()
+    installed_packs = get_installed_packs()
+    playbooks = get_custom_playbooks()
+    automations = get_custom_automations()
+    playbook_stats = get_playbook_stats(playbooks, MAX_DAYS, MAX_REQUEST_SIZE)
+
+    reports = get_custom_reports()
+    dashboards = get_custom_dashboards()
+
+    d = Document(
+        MD_DOCUMENT_TEMPLATE,
+        system_config=system_config,
+        integrations_table=integrations,
+        installed_packs_table=installed_packs,
+        playbooks_table=playbooks,
+        automations_table=automations,
+        open_incidents=open_incidents,
+        closed_incidents=closed_incidents,
+        playbook_stats=playbook_stats,
+        reports=reports,
+        dashboards=dashboards
+    )
+    fr = fileResult("asbuilt.html", d.html())
```

Comment:
since this is a report generated by XSOAR, it should be an "info file"
```suggestion
    fr = fileResult("asbuilt.html", d.html(), file_type=EntryType.ENTRY_INFO_FILE)
```